models/extractor.py

```diff
@@ -89,12 +92,21 @@ def generate_csv(self, entities: Dict, results: List[Any]):
         output.seek(0)
         return output
 
+    def get_similar_docs(self, db: Any, k: int = 5) -> List[Document]:
+        return db.similarity_search(self.question, k)
+
+    async def async_run(self, doc: Document, question: str) -> str:
+        return await self.chain.arun(context=doc.page_content, question=question)
+
+    async def async_run_docs(self, docs: List[Document], question: str) -> List[str]:
+        tasks = [self.async_run(doc, question) for doc in docs]
+        return await asyncio.gather(*tasks)
+
     def run(self, docs: List[Document]) -> List[ExtractionResult]:
         output = []
-        for doc in docs:
-            question = self._get_question()
-            result = self.chain.run(context=doc.page_content, question=question)
+        results = asyncio.run(self.async_run_docs(docs, self.question))
 
+        for result, doc in zip(results, docs, strict=True):
```

Comment:
Interesting, `strict` was [introduced](https://peps.python.org/pep-0618/) in python 3.10 and I'm on 3.9.x. Is there a way to assert a python version for a project besides the readme?

Reply:
Ah gotcha, let me remove to support back compat

Reply:
Idk if there is easy way to python version check. Lets revisit later potentially